### PR TITLE
Make Analyzer crashes a warning

### PR DESCRIPTION
**Pull request by ericstj**
Fixes #78412

@stephentoub  - you mentioned: 
> warnings fail the build and there's nothing a developer can do to address them short of disabling the rule (if any are non-deterministic). We can try it once we're clean, though, and see how it goes.

If new warnings are introduced we can add a NoWarn to the project and file an issue.  Agreed that non-deterministic warnings will require undoing this.   Hopefully we can track such a thing against an issue and undo it with fixed.

To get clean I suppressed this warning and filled the following issues:
https://github.com/dotnet/runtime/issues/90356
https://github.com/dotnet/runtime/issues/90357

### CODE REVIEW

**Comment by stephentoub** [comment]
> To get clean I suppressed this warning and filled the following issues:

I don't see any suppressions. Did you maybe miss a commit?

**Comment by ericstj** [comment]
> > To get clean I suppressed this warning and filled the following issues:
> 
> I don't see any suppressions. Did you maybe miss a commit?

Right you are 🤦‍♂️I waited to build clean and filed all the issues but forgot to commit the suppressions.

**Review by danmoseley — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Looks like there are more failures.

**Comment by ericstj** [comment]
> Looks like there are more failures.

Yeah I figured.  I got a clean Windows Libs build w/ CoreCLR but other flavors need suppressions.

File: src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -46,7 +46,8 @@
     <!-- Override InformationalVersion during servicing as it's returned via public api. -->
     <InformationalVersion Condition="'$(PreReleaseVersionLabel)' == 'servicing'">$(ProductVersion)</InformationalVersion>
     <InformationalVersion Condition="'$(StabilizePackageVersion)' == 'true'">$(ProductVersion)</InformationalVersion>
-    <NoWarn>$(NoWarn),0419,0649,CA2249,CA1830</NoWarn>
+    <!-- AD0001 : https://github.com/dotnet/runtime/issues/90356 -->
+    <NoWarn>$(NoWarn),0419,0649,CA2249,CA1830;AD0001</NoWarn>
```
Comment (stephentoub) [inline src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:50]: Should such a NoWarn be centralized in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems?

Comment (ericstj) [reply src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:50]: I considered it but the generator was referenced in each project and it made more sense to put this in the same spot the generator was referenced.

**Pull request merged**